sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/InstrumentProcessor.java

```diff
@@ -37,41 +37,40 @@
   private final boolean delta;
 
   /**
-   * Create a InstrumentAccumulator that uses the "cumulative" Temporality and uses all labels for
-   * aggregation. "Cumulative" means that all metrics that are generated will be considered for the
-   * lifetime of the Instrument being aggregated.
-   */
-  static <T> InstrumentProcessor<T> getCumulativeAllLabels(
-      InstrumentDescriptor descriptor,
-      MeterProviderSharedState meterProviderSharedState,
-      MeterSharedState meterSharedState,
-      Aggregator<T> aggregator) {
-    return new InstrumentProcessor<>(
-        descriptor,
-        aggregator,
-        meterProviderSharedState.getResource(),
-        meterSharedState.getInstrumentationLibraryInfo(),
-        meterProviderSharedState.getClock(),
-        /* delta= */ false);
-  }
-
-  /**
-   * Create a InstrumentAccumulator that uses the "delta" Temporality and uses all labels for
-   * aggregation. "Delta" means that all metrics that are generated are only for the most recent
+   * Create a new {@link InstrumentProcessor} for use in metric recording aggregation.
+   *
+   * <p>"Delta" temporality means that all metrics that are generated are only for the most recent
    * collection interval.
+   *
+   * <p>"Cumulative" temporality means that all metrics that are generated will be considered for
+   * the lifetime of the Instrument being aggregated.
    */
-  static <T> InstrumentProcessor<T> getDeltaAllLabels(
-      InstrumentDescriptor descriptor,
+  static <T> InstrumentProcessor<T> createProcessor(
       MeterProviderSharedState meterProviderSharedState,
       MeterSharedState meterSharedState,
-      Aggregator<T> aggregator) {
-    return new InstrumentProcessor<>(
-        descriptor,
-        aggregator,
-        meterProviderSharedState.getResource(),
-        meterSharedState.getInstrumentationLibraryInfo(),
-        meterProviderSharedState.getClock(),
-        /* delta= */ true);
+      InstrumentDescriptor descriptor,
+      AggregationConfiguration configuration) {
+    Aggregator<T> aggregator =
+        configuration.getAggregatorFactory().create(descriptor.getValueType());
+    switch (configuration.getTemporality()) {
+      case CUMULATIVE:
+        return new InstrumentProcessor<>(
+            descriptor,
+            aggregator,
+            meterProviderSharedState.getResource(),
+            meterSharedState.getInstrumentationLibraryInfo(),
+            meterProviderSharedState.getClock(),
+            /* delta= */ false);
+      case DELTA:
+        return new InstrumentProcessor<>(
+            descriptor,
+            aggregator,
+            meterProviderSharedState.getResource(),
+            meterSharedState.getInstrumentationLibraryInfo(),
+            meterProviderSharedState.getClock(),
+            /* delta= */ true);
+    }
```

Comment:
The only difference between the two cases is the delta boolean.  If you make a local var like 
```
boolean delta = configuration.getTemporality() == DELTA;
```
you can simplify the switch, remove the `/* delta */` comment, and it cleans up a little bit.

Reply:
Good suggestion, done.